added info link for Plugin Health Score

Harsh3341 opened the pull request:
https://github.com/jenkins-infra/plugin-site/pull/1320#issuecomment-1503630565

Info links for plugins is been added

Harsh3341 commented:
Solved #1328

NotMyFault commented on plugins/plugin-site/src/components/PluginHealthScore.jsx:
```diff
@@ -1,15 +1,43 @@
+                        <a
+                            href={`https://plugin-health.jenkins.io/scores/${name}`}
+                        >
+                            here
```
```suggestion
                        View the plugin's
                        {' '}
                        <a
                            href={`https://plugin-health.jenkins.io/scores/${name}`}
                        >
                            total health score.
```
Using words like 'here' for URLs isn't a good practice, see https://uxmovement.com/content/why-your-links-should-never-say-click-here/, nor do we need to state that this is the health score, given it does already say so right below the the box.

I recommend moving this to a dedicated element because clicking on the question mark links through the plugin page, rather than directly to the health score page, causing unneeded browser noise.

Harsh3341 replied:
Thanks for informing me about this good practice, and I think I should remove the link from the question mark because I already added a tooltip

Harsh3341 committed:
added info link for Plugin Health Score

NotMyFault commented:
Please don't mark comments as resolved when they are not. 

> I recommend moving this to a dedicated element because clicking on the question mark links through the plugin page, rather than directly to the health score page, causing unneeded browser noise.

is still to address.

I'd recommend keeping the question mark clickable, additionally to the hover.

Harsh3341 commented:
> I'd recommend keeping the question mark clickable, additionally to the hover.

It is clickable and also can be hovered

NotMyFault commented:
> > I'd recommend keeping the question mark clickable, additionally to the hover.
> 
> It is clickable

Why do you think so? You removed the logic for that, after my previous comment, for no reason.

Harsh3341 commented:
> How did you test this? On the preview site clicking the question mark does nothing.
![IMG_20230413_121128](https://user-images.githubusercontent.com/101264150/231675491-fb827da1-1926-4a35-885d-8718f34e53cc.jpg)

Currently I'm in my phone while clicking on "?" Tooltip is opening, also I have tested this on desktop while clicking and hover

NotMyFault commented:
I didn't mention the tooltip, I mentioned the linking action you previously had on the question mark. The need to hover first and click later just requires more action of the user, when there's no need for it.

Harsh3341 commented:
> I didn't mention the tooltip, I mentioned the linking action you previously had on the question mark. The need to hover first and click later just requires more action of the user, when there's no need for it.

O.. I misunderstood what you said before, but now I get it.

What I have done is I have removed link from "?" and added `stopPropagation()` when clicked on "?" So that it does not redirect to plugin page and also I have removed link because for phone users I want toolkit to open onclick rather then directly redirecting to Health Score page. 

What needs to be done is I have to add `stopPropagation()` in tooltip `health score`

NotMyFault commented:
> What I have done is I have removed link from "?" and added `stopPropagation()` when clicked on "?" So that it does not redirect to plugin page and also I have removed link because for phone users I want toolkit to open onclick rather then directly redirecting to Health Score page.

Yeah, I questioned the "why". This design causes more noise for the end user, especially if the hover is not sticky, if you're not fast quick, you don't make it:

https://user-images.githubusercontent.com/13383509/231681630-e58c0faf-1a22-4cd6-a9d1-52c01c8ebee6.mp4

Harsh3341 commented:
Noted...

With all fixes, I will push the next commit

Harsh3341 commented:
> Yeah, I questioned the "why". This design causes more noise for the end user, especially if the hover is not sticky, if you're not fast quick, you don't make it:

just now checked the code its reactstrap tooltip [component ](https://reactstrap.github.io/?path=/docs/components-tooltip--tooltip) causing that issue of hover, tested some changes but still result remains the same, so should I remove the tooltip component and rewrite a different code for it?

NotMyFault commented:
> > Yeah, I questioned the "why". This design causes more noise for the end user, especially if the hover is not sticky, if you're not fast quick, you don't make it:
> 
> just now checked the code its reactstrap tooltip [component ](https://reactstrap.github.io/?path=/docs/components-tooltip--tooltip) causing that issue of hover, tested some changes but still result remains the same, so should I remove the tooltip component and rewrite a different code for it?

None of that is needed, and out of the scope of the PR. All I asked for is re-adding the ability to click the question mark.

Harsh3341 committed:
re-added the ability to click the question mark

NotMyFault reviewed (approved):
Thanks!

halkeye commented on plugins/plugin-site/src/components/PluginHealthScore.jsx:
```diff
@@ -24,6 +57,7 @@ PluginHealthScore.propTypes = {
     healthScore: PropTypes.shape({
         value: PropTypes.number,
```
```suggestion
        value: PropTypes.number.isRequired,
```

halkeye commented on plugins/plugin-site/src/components/PluginHealthScore.jsx:
```diff
@@ -24,6 +57,7 @@ PluginHealthScore.propTypes = {
     healthScore: PropTypes.shape({
         value: PropTypes.number,
     }),
+    name: PropTypes.string,
```
you don't handle name not being provided, so it should be required
```suggestion
    name: PropTypes.string.isRequired,
```

halkeye replied:
@Harsh3341 do you want to do a followup pr with these? I'm just going to merge.

halkeye committed:
Merge branch 'master' into info-about-health-score
